engine/src/main/java/org/camunda/bpm/engine/impl/db/sql/DbSqlSession.java

```diff
@@ -302,7 +304,34 @@ public static boolean isCrdbConcurrencyConflict(Throwable cause) {
     return false;
   }
 
+  /**
+   * In cases where CockroachDB is used, and a failed operation is detected,
+   * the method checks if the exception was caused by a CockroachDB
+   * <code>TransactionRetryException</code>. This method may be used when a
+   * CRDB Error occurs on commit, and a Command Context is not available, as
+   * it has already been closed. This is the case with Spring/JTA transaction
+   * interceptors.
+   *
+   * @param cause for which an operation failed
+   * @param configuration of the Process Engine
+   * @return true if the failure was due to a CRDB <code>TransactionRetryException</code>.
+   *          Otherwise, it's false.
+   */
+  public static boolean isCrdbConcurrencyConflictOnCommit(Throwable cause, ProcessEngineConfigurationImpl configuration) {
+    // only check when CRDB is used
+    if (DatabaseUtil.checkDatabaseType(configuration, DbSqlSessionFactory.CRDB)) {
+      // with externally managed transactions, the real cause is sometimes suppressed
+      List<Throwable> causes = new ArrayList<>(Arrays.asList(cause.getSuppressed()));
+      causes.add(cause);
```

Comment:
This part is when JTA is used, since it suppresses the CRDB error and provides a `RollbackException` instead, so the real cause is not detected. I will expand on the explanation that it is only for JTA.